File: mns-core/src/test/java/cz/fb/manaus/core/model/MarketSnapshotTest.java
package cz.fb.manaus.core.model;

import org.apache.commons.lang3.time.DateUtils;
import org.junit.Test;

import java.util.Date;
import java.util.List;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class MarketSnapshotTest {

    @Test
    public void testMarketCoverage() {
        var selectionId = 1L;
        var side = Side.LAY;
        var predecessor = new Bet(null, null, selectionId, new Price(2d, 2d, side), DateUtils.addHours(new Date(), -2), 1d);
        var successor = new Bet(null, null, selectionId, new Price(2d, 2d, side), new Date(), 1d);
        var coverage = MarketSnapshot.getMarketCoverage(List.of(successor, predecessor));
        assertThat(coverage.size(), is(1));
        assertThat(coverage.get(side, selectionId), is(successor));
    }

}

File: mns-core/src/test/java/cz/fb/manaus/core/model/MarketSnapshotTest.kt
package cz.fb.manaus.core.model

import org.junit.Test
import java.time.Instant
import java.time.temporal.ChronoUnit
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class MarketSnapshotTest {

    @Test
    fun `market coverage`() {
        val selectionId = 1L
        val side = Side.LAY
        val predecessor = Bet(
                marketId = "1",
                selectionId = selectionId,
                requestedPrice = Price(2.0, 2.0, side),
                placedDate = Instant.now().minus(2, ChronoUnit.HOURS),
                matchedAmount = 1.0
        )
        val successor = predecessor.copy(placedDate = Instant.now())
        val coverage = getMarketCoverage(listOf(successor, predecessor))
        assertEquals(1, coverage.size)
        assertEquals(successor, coverage[SideSelection(side, selectionId)])
    }

    @Test
    fun `is active`() {
        val bet = Bet(
                marketId = "1",
                selectionId = 1L,
                requestedPrice = Price(2.0, 2.0, Side.LAY),
                placedDate = Instant.now().minus(2, ChronoUnit.HOURS),
                matchedAmount = 1.0
        )
        val coverage = getMarketCoverage(listOf(bet))
        assertTrue { coverage.isActive(bet.selectionId) }
        assertFalse { coverage.isActive(bet.selectionId + 100) }
    }

}